engine/execution/computation/computer/computer.go

```diff
@@ -332,31 +328,35 @@ func (e *blockComputer) executeTransaction(
 	res *execution.ComputationResult,
 ) error {
 	startedAt := time.Now()
-	var txSpan opentracing.Span
-	var traceID string
-	// call tracing
-	txSpan = e.tracer.StartSpanFromParent(colSpan, trace.EXEComputeTransaction)
+	txID := txBody.ID()
 
-	if sc, ok := txSpan.Context().(jaeger.SpanContext); ok {
-		traceID = sc.TraceID().String()
-	}
+	// we capture two spans one for tx-based view and one for the current context (block-based) view
+	txSpan := e.tracer.StartSpanFromParent(colSpan, trace.EXEComputeTransaction)
+	txSpan.LogFields(log.String("tx_id", txID.String()))
+	txSpan.LogFields(log.Uint32("tx_index", txIndex))
+	txSpan.LogFields(log.Int("col_index", collectionIndex))
+	defer txSpan.Finish()
 
-	defer func() {
-		txSpan.LogFields(
-			log.String("transaction.ID", txBody.ID().String()),
-		)
-		txSpan.Finish()
-	}()
+	var traceID string
+	txInternalSpan, _, isSampled := e.tracer.StartTransactionSpan(context.Background(), txID, trace.EXERunTransaction)
+	if isSampled {
+		txInternalSpan.LogFields(log.String("tx_id", txID.String()))
+		if sc, ok := txInternalSpan.Context().(jaeger.SpanContext); ok {
+			traceID = sc.TraceID().String()
+		}
+	}
+	defer txInternalSpan.Finish()
 
 	e.log.Debug().
 		Hex("tx_id", logging.Entity(txBody)).
 		Msg("executing transaction")
 
-	txView := collectionView.NewChild()
-
 	tx := fvm.Transaction(txBody, txIndex)
```

Comment:
Don't we want to move this below trace span as well?

Reply:
not sure if I understand? you mean the NewChild() call?

Reply:
SetTraceSpan won't start a trace, it would only pass the parent span reference to transaction object